### DISCUSSION TITLE
CSS-like transitions

### DISCUSSION THREAD

**Discussion by LukeTOBrien** [dotnet/maui#309] [Ideas]
Hello there,

I am interested in building an app for MAUI and I have recently started creating my new project in WPF, assuming that MAUI will too be XAML I can port my app when it comes to it.  
I come from a mainly web background and although I love XAML I have not had a chance to do any.

I have a grid layout with two row definitions with their height bound to the `State` property of an internal object (using an IValueConverter).  
So when in `Hidden` state the rows adjust their height.  
[In CSS I could do](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Transitions/Using_CSS_transitions) `transition: height 2s` but in XAML I have to create a Storyboard and there is no easy way to animate `RowDefinition`, I have come across [this SO answer](https://stackoverflow.com/questions/46771899/how-to-animate-grid-row-height).

So I suggest perhaps:
```xaml
<Grid.Transitions>
   <Transition x:Name="Slide" PropertyName="Height" Duration="2s"/>
<Grid.Transitions>
```
Or in my binding case:
```xaml
<RowDefinition Height="{Binding State, Converter={StaticResource StateToGridLength}, Transition="Slide"}" />
```

I just wanted to put this down while it's in my head.  
What do you think?

Is MAUI even going to be XAML? I assume it is and will it have the same features as exist in WPF?  
I understand Xamarin.Forms is slightly different to WPF.

**Comment by mrwcjoughin** [comment]
MAUI is going to use XAML - and its Xamarin.Forms' XAML not WPF or UWP's

the only upgrade path I have heard of is from Xamarin.Forms to MAUI - not
sure if they will offer one for WPF to MAUI
